### PR TITLE
Add LINQ method, edit method, ShowTop

### PR DESCRIPTION
**Pull request by VasylKits**

### CODE REVIEW

File: AnnouncementWebApi/Models/EditAnnouncement.cs [hunk 1]
```diff
@@ -7,5 +7,10 @@ public class EditAnnouncement
         public int Id { get; set; }
         public string Title { get; set; }
         public string Description { get; set; }
+        public DateTime EditDate { get; set; }
```
Comment (nikoliasoroka) [inline AnnouncementWebApi/Models/EditAnnouncement.cs:10]: Why do we need the EditDate property for the DTO model, when we don't have it for the main entity Announcement? What is the goal of this property?

File: AnnouncementWebApi/Controllers/AnnouncementController.cs [hunk 2]
```diff
@@ -34,8 +34,10 @@ public IActionResult AddAnnouncement([FromBody] NewAnnouncement newAnnouncement)
         }
 
         [HttpPut]
-        public IActionResult EditAnnouncement(EditAnnouncement editAnnouncement)
+        public IActionResult EditAnnouncement([FromBody]EditAnnouncement editAnnouncement)
         {
+            //Announcement editAnn = announcements.SingleOrDefault(a => a.Id == editAnnouncement.Id);
+            //editAnn = new() { Title = editAnnouncement.Description, Description = editAnnouncement.Description };
```
Comment (nikoliasoroka) [inline AnnouncementWebApi/Controllers/AnnouncementController.cs:40]: When you got the announcement by Id, you just should update his properties (editAnn.Title = editAnnouncement.Title and so on...) and that's it.

File: AnnouncementWebApi/Models/EditAnnouncement.cs (hunk 1 above)
Comment (nikoliasoroka) [reply AnnouncementWebApi/Models/EditAnnouncement.cs:10]: Leave your answer here in the comments, please.

File: AnnouncementWebApi/Controllers/AnnouncementController.cs (hunk 2 above)
Comment (VasylKits) [reply AnnouncementWebApi/Controllers/AnnouncementController.cs:40]: How I can write and save this changes(editAnn.Title = editAnnouncement.Title) into new object? Does method Edit includes one input param or two(int id, EditAnnouncement editAnnouncement)?

File: AnnouncementWebApi/Models/EditAnnouncement.cs (hunk 1 above)
Comment (VasylKits) [reply AnnouncementWebApi/Models/EditAnnouncement.cs:10]: The EditDate property was added when the announcement has been edited. But I don't know where I have to announce it...

File: AnnouncementWebApi/Controllers/AnnouncementController.cs (hunk 2 above)
Comment (nikoliasoroka) [reply AnnouncementWebApi/Controllers/AnnouncementController.cs:40]: When you assign the property editAnn.Title = editAnnouncement.Title it already saved.
Should include the EditAnnouncement parameter only, it already has an Id.
Use debugging, please, and you will see what you do!!!

File: AnnouncementWebApi/Models/EditAnnouncement.cs (hunk 1 above)
Comment (nikoliasoroka) [reply AnnouncementWebApi/Models/EditAnnouncement.cs:10]: If you want to add EditDate and mark the time of editing, you should add it to the Announcement class as well. So do it please.

Comment (VasylKits) [reply AnnouncementWebApi/Models/EditAnnouncement.cs:10]: I changed the method EditAnnouncement. I created EditDate property in Announcement class and set default value as DateTime.Now

Comment (nikoliasoroka) [reply AnnouncementWebApi/Models/EditAnnouncement.cs:10]: I think would be better to leave EditDate with NULL when we create an object and populate it when we edit the object only.

File: AnnouncementWebApi/Controllers/AnnouncementController.cs [hunk 3]
```diff
@@ -72,12 +82,9 @@ public IActionResult GetId(int id)
...
 
         [HttpGet("TOP")]
-        public IActionResult ShopTopThreeAnnouncements()
+        public IActionResult ShowTopThreeAnnouncements()
         {
-            var result = from a in announcements
-                         where a.Title.Contains("announcement")
-                         orderby a.Id
-                         select a;
+            var result = announcements.Where(a => a.Title.Contains("announcement")).GroupBy(a => a.Id);
```
Comment (nikoliasoroka) [inline AnnouncementWebApi/Controllers/AnnouncementController.cs:87]: What is the goal of using the GroupBy() method?

**Pull request merged**